arch/x86_64/include/arch.h

```diff
@@ -0,0 +1,92 @@
+/****************************************************************************
+ * arch/x86_64/include/arch.h
+ *
+ *   Copyright (C) 2011 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```

Comment:
@sonicyang All new files added to the repository must have have Apache 2.0 headers with no copyright or author information.  you can copy the header from sched/sched/sched_getcpu.c for example.

For those files that have my copyright on them, I do grant you permission to change the license to the Apache 2.0 license.

Sorry but those are Apache rules:  All new files must have the Apache 2.0 license.  Older files will eventually be converted to Apache 2.0 license as well, but that will require more time.

In general, the code looks good.  Have you ran tools/nxstyle against it?  No code can come into the repositories that does not meet the NuttX coding standard.  The nxstyle program will do some basic checks for conformance to the codeing standard.

If you do those two things, I will be happy to merge the code into the repository.

Reply:
This is a large change and difficult to review online.  Once you complete the above changes, I will merge to a branch were I can do a more thorough review.  Assume there are additional issues, I will then merge that branch to master.